### ISSUE TITLE
[Bug]: Using uploaded SmartModules returns error:  0: Is a directory (os error 21)

### ISSUE THREAD

**Issue by ajhunyady** [infinyon/fluvio#2292]
In glucose demo, I created the following `SmartModule`:

```
use fluvio_smartmodule::{smartmodule, Record, RecordData, Result};

#[derive(serde::Deserialize)]
#[serde(rename_all = "camelCase")]
struct FullRecord {
    event_type: String,
    source_device_id: String,
    blood_glucose_value: u32,
    pub transmitter_id: String,
    geolocation: String,
    timestamp: i64,
}

#[smartmodule(filter_map)]
pub fn filter_map(record: &Record) -> Result<Option<(Option<RecordData>, RecordData)>> {
    let full_record = serde_json::from_slice::<FullRecord>(record.value.as_ref())?;

    if full_record.blood_glucose_value > 190 {
        let reading = format!("Device `{}` - high glucose: `{}`, timestamp `{}`", 
            full_record.transmitter_id, 
            full_record.blood_glucose_value,
            full_record.timestamp
        );
        Ok(Some((None, RecordData::from(reading))))
    } else {
        Ok(None)
    }
}
```

The SmartModule works well when invoked from the cli:

```
% fluvio consume glucose --filter-map filter-map-slack/target/wasm32-unknown-unknown/release/filter_map_slack.wasm -T=100
Consuming records starting 100 from the end of topic 'glucose'
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439262131`
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439263131`
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439275131`
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439283131`
Device `ok6j44yai5` - high glucose: `199`, timestamp `1649439296131`
Device `ok6j44yai5` - high glucose: `196`, timestamp `1649439300131`
```

But fails if I load and invoke the SM module I uploaded.

```
% fluvio sm create filter-map-slack --wasm-file filter-map-slack/target/wasm32-unknown-unknown/release/filter_map_slack.wasm
% fluvio consume glucose --filter-map filter-map-slack -T=100 
Error: 
   0: Is a directory (os error 21)
```

**Comment by morenol** [comment]
Could you try for now with a name for the smartmodule that does not match with a folder name in the current directory?

**Comment by ajhunyady** [comment]
that works!

% fluvio sm create fm-slack --wasm-file filter-map-slack/target/wasm32-unknown-unknown/release/filter_map_slack.wasm        
% fluvio consume glucose --filter-map fm-slack -T=100
Consuming records starting 100 from the end of topic 'glucose'
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439262131`
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439263131`
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439275131`
Device `ok6j44yai5` - high glucose: `191`, timestamp `1649439283131`
Device `ok6j44yai5` - high glucose: `199`, timestamp `1649439296131`
Device `ok6j44yai5` - high glucose: `196`, timestamp `1649439300131`

But still needs to be fixed as it was not obvious why it failed.

**Comment by ajhunyady** [comment]
@morenol, thanks for the quick turnaround on finding a solution.

**Comment by morenol** [comment]
So, the issue is that when passing an smartmodule, it first checks if a file with that name exists locally and if that is the case use that, otherwise use named smartmodules. 

When there is a directory with the same name of the smartmodule, the consumer tries to use that but it fails.

In that case, we should detect that it is a directory and proceed with named smartmodules instead.

**Comment by morenol** [comment]
CLosed by #2328
